samples/SimpleDataSource/Tables/WordTable.cs

```diff
@@ -0,0 +1,138 @@
+        {
+            //
+            // Implement your columns here. 
+            // Columns are implemented via Projections, which are simply functions that map a row index to a data point.
+            //
+
+            //
+            // Pre-process the data to get a list of <filename, time, word> tuples.
+            //
+
+            var allWords = new List<Tuple<string, Timestamp, string>>();
+            foreach(var kvp in this.Lines)
+            {
+                foreach (var tuple in kvp.Value.SelectMany(x => x.Item2.Split((char[])null, StringSplitOptions.RemoveEmptyEntries).Select(y => new Tuple<Timestamp, string>(x.Item1, y))))
+                {
+                    allWords.Add(Tuple.Create(kvp.Key, tuple.Item1, tuple.Item2));
+                }
+            }
+
+            //
+            // Use Projection.Index() to get a base projection from row index to a <filename, word> tuple.
+            //
+
+            var baseProjection = Projection.Index(allWords);
+
+            //
+            // Create projection for each column by composing the base projection with another projection that maps to the data point as needed.
+            //
+
+            var fileProjection = baseProjection.Compose(x => x.Item1);
+            var wordProjection = baseProjection.Compose(x => x.Item3);
+            var charCountProjection = baseProjection.Compose(x => x.Item3.Length);
+            var timeProjection = baseProjection.Compose(x => x.Item2);
+
+            //
+            // Table Configurations describe how your table should be presented to the user: 
+            // the columns to show, what order to show them, which columns to aggregate, and which columns to graph. 
+            // You may provide a number of columns in your table, but only want to show a subset of them by default so as not to overwhelm the user. 
+            // The table configuration class also exposes four (4) columns that viewers can recognize: Pivot Column, Graph Column, Left Freeze Column, Right Freeze Column
+            // For more information about what these columns do, go to "Advanced Topics" -> "Table Configuration" in our Wiki. Link can be found in README.md
+            //
+
+            var config = new TableConfiguration("Word Time")
+            {
+                Columns = new[]
+              {
+                    WordColumn,
+                    TableConfiguration.PivotColumn,
+                    FileColumn,
+                    CharacterCountColumn,
+                    TableConfiguration.GraphColumn,
+                    TimeColumn,
+                },
+                Layout = TableLayoutStyle.GraphAndTable,
+            };
+
+            //
+            //
+            //  Use the table builder to build the table. 
+            //  Add and set table configuration if applicable.
+            //  Then set the row count (we have one row per file) and then add the columns using AddColumn.
```

Comment:
one row per file sounds incorrect, `allWords.Count` will be the number of words across all files, right?

Reply:
yeah, its row per a word